Dockerfile

```diff
@@ -0,0 +1,7 @@
+FROM golang:1.23.2
+
+WORKDIR /app
+
+COPY . .
```

Comment:
To best utilize caching layers, we should copy in the mod files first, go `go get` and then copy in the rest.